pkg/btp/Authentication.go

```diff
@@ -0,0 +1,96 @@
+package btp
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+type BTPUtils struct {
+	Exec     ExecRunner
+	loggedIn bool
+}
+
+type LoginOptions struct {
+	Url       string
+	Subdomain string
+	User      string
+	Password  string
+	Tenant    string
+}
+
+func (btp *BTPUtils) LoginCheck() (bool, error) {
+	return btp.loggedIn, nil
+}
+
+func (btp *BTPUtils) Login(options LoginOptions) error {
+	var err error
+
+	_r := btp.Exec
+
+	if _r == nil {
+		_r = &Executer{}
+	}
+
+	if options.Url == "" || options.Subdomain == "" || options.User == "" || options.Password == "" {
+		return fmt.Errorf("Failed to login to BTP: %w", errors.New("Parameters missing. Please provide the CLI URL, Subdomain, Space, User and Password"))
+	}
+
+	var loggedIn bool
+
+	loggedIn, err = btp.LoginCheck()
+
+	if loggedIn == true {
+		return err
```

Comment:
Why does this function return an error when the btp cli is already logged in?
How should the caller handle this error?